lib/widgets/lightbox.dart

```diff
@@ -197,22 +199,207 @@ class _LightboxPageState extends State<_LightboxPage> {
+    final store = PerAccountStoreWidget.of(context);
+    final zulipLocalizations = ZulipLocalizations.of(context);
+
+    assert(debugLog('VideoPlayerController.networkUrl(${widget.src})'));
+    _controller = VideoPlayerController.networkUrl(widget.src, httpHeaders: {
+      if (widget.src.origin == store.account.realmUrl.origin) ...authHeader(
+        email: store.account.email,
+        apiKey: store.account.apiKey,
+      ),
+      ...userAgentHeader()
+    });
+    _controller!.addListener(_handleVideoControllerUpdates);
+
+    try {
+      await _controller!.initialize();
+      await _controller!.play();
+    } catch (error) { // TODO(log)
+      assert(debugLog("VideoPlayerController.initialize failed: $error"));
+      if (mounted) {
+        await showErrorDialog(
+          context: context,
+          title: zulipLocalizations.errorDialogTitle,
+          message: zulipLocalizations.errorVideoPlayerFailed,
+          // To avoid showing the disabled video lightbox for the unnsupported
+          // video, we make sure user doesn't reach there by dismissing the dialog
+          // by clicking around it, user must press the 'OK' button, which will
+          // take user back to content message list.
+          barrierDismissible: false,
+          onContinue: () {
+            Navigator.pop(context); // Pops the dialog
+            Navigator.pop(context); // Pops the lightbox
+          });
+      }
+    }
+  }
+
+  @override
+  void dispose() {
+    _deinitialize();
+    super.dispose();
+  }
+
+  void _deinitialize() {
+    _controller?.removeListener(_handleVideoControllerUpdates);
+    _controller?.dispose();
+    _controller = null;
+  }
+
+  void _handleVideoControllerUpdates() {
+    setState(() {});
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return _LightboxPageLayout(
+      routeEntranceAnimation: widget.routeEntranceAnimation,
+      message: widget.message,
+      buildBottomAppBar: (context, color, elevation) =>
+        _controller == null
+          ? null
+          : BottomAppBar(
+              height: 150,
+              color: color,
+              elevation: elevation,
+              child: Column(mainAxisAlignment: MainAxisAlignment.end, children: [
+              Row(children: [
+                Text(_formatDuration(_controller!.value.position),
+                  style: const TextStyle(color: Colors.white)),
+                Expanded(child: Slider(
+                  value: _controller!.value.position.inSeconds.toDouble(),
+                  max: _controller!.value.duration.inSeconds.toDouble(),
+                  activeColor: Colors.white,
+                  onChanged: (value) {
+                    _controller!.seekTo(Duration(seconds: value.toInt()));
+                  })),
+                Text(_formatDuration(_controller!.value.duration),
+                  style: const TextStyle(color: Colors.white)),
+                ]),
+              IconButton(
```

Comment:
A line starting with close-bracket (or close-paren, etc.) should be indented the same as the line with the corresponding open-bracket (or open-parent, etc.).

So this `]),` should match the indentation of `Row(children: [`, rather than be indented an extra level. Combined with the comment above (the `Row` should be indented one more level), that comes to:

```suggestion
                  Text(_formatDuration(_controller!.value.duration),
                    style: const TextStyle(color: Colors.white)),
                ]),
                IconButton(
```